syfertext/vocab.py

```diff
@@ -18,5 +28,213 @@ def __init__(self, model_name: str):
+        # with key as correseponding attribute ID.
+        self.lex_attr_getters = LEX_ATTRS
+
+        # List of stop words
+        self.stop_words = STOP_WORDS
+
+        # get the stop word attribute getter.
+        is_stop = self.lex_attr_getters[Attributes.IS_STOP]
+
+        # Update the function for is_stop with the stop word list
+        self.lex_attr_getters[Attributes.IS_STOP] = functools.partial(
+            is_stop, stops=self.stop_words
+        )
+
         # Create the Vectors object
-        self.vectors = Vectors(self.model_name)
+        self.vectors = Vectors(model_name)
+
+    def load_strings(self) -> List[str]:
+        """load the pickled list of words that the Vocab object knows and has vectors for"""
+
+        words_path = os.path.join(self.model_path, "words")
+
+        with open(words_path, "rb") as word_file:
+            strings = pickle.load(word_file)
+
+        return strings
+
+    def get_vector(self, key: Union[str, int]):
+        """Retrieve a vector for a word in the vocabulary. Words can be looked
+        up by string or int ID. 
+
+        Args:
+            key: The word hash, or its plaintext.
+        """
+        if isinstance(key, int):
+            key = self.store[key]
+
+        return self.vectors[key]
+
+    def __iter__(self):
+        """Iterate over the lexemes in the vocabulary.
+
+        Yields:
+            lexeme (Lexeme): An entry in the vocabulary.
+        """
+
+        for orth in self.lex_store:
+
+            # Create the Lexeme object for the given orth
+            lexeme = Lexeme(self, orth)
+
+            yield lexeme
+
+    def __getitem__(self, key: Union[str, int]) -> Lexeme:
+        """Retrieve a Lexeme object corresponding to a string, given its hash or its plaintext. If a
+        previously unseen string is given, a new lexeme is created and
+        stored.
+
+        Args:
+            key: The word hash, or its plaintext. 
+
+        Returns:
+            Lexeme: The lexeme specified by the given key.
+        """
+
+        if isinstance(key, str):
+            orth = self.store.add(key)
+
+        else:
+            orth = key
+
+        return Lexeme(self, orth)
+
+    def __contains__(self, key: Union[str, int]) -> bool:
+        """Check whether the lexeme for the string or int key has an entry in the vocabulary."""
+
+        if isinstance(key, str):
+            orth = self.store[key]
+
+        else:
+            orth = key
+
+        return orth in self.lex_store
+
+    def has_vector(self, key: Union[str, int]) -> bool:
+        """Check whether the given string has an entry in word vectors table
+        
+        Args:
+            key: The word or its hash for which the existence of a vector is to be checked out.
+        
+        Returns :
+            bool: True if a vector for 'word' already exists.
+        """
+
+        return self.vectors.has_vector(key)
+
+    def add_flag(self, flag_getter: Callable, flag_id=-1) -> int:
```

Comment:
I actually suggest to remove this method. We do not have a mature way for serializing a callable for now. Plans in PySyft are under construction, but they are not going to support arbitrary computations before a while. I advanced this argument to @sachin-101 regarding callbacks in Matchers.

Reply:
If it turns out that more flags are needed, we can add them to the build-in attributes.

Reply:
Until plans are mature enough

Reply:
okay